### ISSUE TITLE
Travis uses old clang version

### ISSUE THREAD

**Issue by FrancescoBorzi** [azerothcore/azerothcore-wotlk#1259]
In travis the installed clang version is 7.0.0:

```
clang --version
clang version 7.0.0 (tags/RELEASE_700/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/clang-7.0.0/bin
```

However the build for some reasons uses an older version (3.8.0):

```
-- The C compiler identification is Clang 3.8.0
-- The CXX compiler identification is Clang 3.8.0
```

--------

This causes some build fail because of some old clang bugs that were fixed in newer versions.

For example, this build:

https://travis-ci.org/azerothcore/azerothcore-wotlk/builds/478049211

Is failing because of this bug (resolved in clang 4.8+) :
https://bugs.llvm.org/show_bug.cgi?id=21629

**Comment by BarbzYHOOL** [comment]
Wait, so it means we need clang 7.0? you know for debian for example it's in debian 10 (unreleased) and it's not backported, while the stable clang for 4 years is 3.8

**Comment by FrancescoBorzi** [comment]
Our build is currently broken with clang 7.0. I'm considering opening a separate issue for that, and use version 6.0.0 in travis. Version 6.0.0 is the default version in Ubuntu 18.04. No need to compile manually anything.

**Comment by FrancescoBorzi** [comment]
I got your point. Travis is in 16.04 but you can install any clang version (on travis, it defaults to 7.0.0 unless you install it via apt instead of using the one provided by travis).

`In top of that, when it comes to server it's more between debian, centos rather than ubuntu`

Agree. How easy is to install a clang version newer than 3.8 in the other distro that you mentioned?

If it's hard and you have to compile etc... then I agree with you and let's keep supporting clang 3.8 and having it in our travis build.

If it's easy (for example in Ubuntu 16.04 is very straightforward to install a newer clang), then I would change the version in travis too.

**Comment by BarbzYHOOL** [comment]
Debian is super stable thanks to this release system.

My bad, I thought there were no backports for Clang but it's because it's named differently:

```
clang-3.8/stable,now 1:3.8.1-24 amd64 [installed,automatic]
clang-3.8-doc/stable 1:3.8.1-24 all
clang-3.8-examples/stable 1:3.8.1-24 amd64
clang-3.9/stable 1:3.9.1-9 amd64
clang-3.9-doc/stable 1:3.9.1-9 all
clang-3.9-examples/stable 1:3.9.1-9 amd64
clang-4.0/stable 1:4.0.1-10~deb9u2 amd64
clang-4.0-doc/stable 1:4.0.1-10~deb9u2 all
clang-4.0-examples/stable 1:4.0.1-10~deb9u2 amd64
clang-5.0/stretch-backports 1:5.0.1-2~bpo9+1 amd64
clang-5.0-doc/stretch-backports 1:5.0.1-2~bpo9+1 all
clang-5.0-examples/stretch-backports 1:5.0.1-2~bpo9+1 amd64
clang-6.0/testing,unstable 1:6.0.1-9.2 amd64
clang-6.0/stretch-backports 1:6.0-1~bpo9+1 amd64
clang-7/testing,unstable 1:7.0.1-4 amd64
```
(no clang 7 though)

So that is not as bad as I thought but it still requires some manipulations and can be more prone to instability (in theory). But I could live that personally xD

On debian 9, most people wouldn't know how to install a newer clang (and might even stay on GCC). We can get the latest (7.0) with apt, but it's not recommended aka frankendebian (mixing packages from debian 10 with debian 9) https://www.debian.org/releases/

----------------
For centos, I think @Viste uses it (but not me). I think centos releases have the biggest longevity ever.

---------------
For ubuntu versions XX.10, it's always a testing version that is supported 6 month/1 year, so it's not even a good benchmark. Only XX.04 are important for our users (in my opinion).16.04 will be supported till 2021, 18.04 is the latest and will be supported till 2023.

**Comment by FrancescoBorzi** [comment]
My suggestion:

- set the travis build on clang 6.0.0
- recommend to the users to use clang 6.0.0 in the wiki (with instructions)
- open a separate issue to fix build for clang 7.0.0 (so Ubuntu 18.10 don't have to downgrade)
- avoid breaking the build on clang 3.8 unless strictly necessary
